src/VisualStudio/Core/Def/Options/VisualStudioSettingsOptionPersister.cs

```diff
@@ -116,6 +116,12 @@ private System.Threading.Tasks.Task OnSettingChangedAsync(object sender, Propert
                     return stringArray.ToImmutableArray();
                 }
 
+                if (optionKey.Option.Type == typeof(int) &&
+                    _settingManager.TryGetValue(storageKey, out int intValue) == GetValueResult.Success)
+                {
+                    return intValue;
+                }
```

Comment:
Should we instead update the following line to match the type stored in `_settingsManager`? https://github.com/dotnet/roslyn/blob/f814f00d41352cac7ed88d745584191e6ce2323e/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/Core/Formatting/FormattingOptions2.cs#L55

Reply:
I don't think so. The fact that the option is stored as In64 in VS setting is a VS-specific quirk. Quite odd, since there is no reason for it to be 64 bit.

Reply:
@tmat Then what about the more generalized approach suggested in https://github.com/dotnet/roslyn/pull/65382#pullrequestreview-1178115501?

This will ensure that no such issues exist whatever the type of option is.

Reply:
Will follow up on that later on: https://github.com/dotnet/roslyn/issues/65394